### DISCUSSION TITLE
Cannot persist login state

### DISCUSSION THREAD

**Discussion by dirums** [swellstores/community#122] [Help] [answered]
Hey, I've been using this awesome platform for a quit while and I'm loving it but one thing I'm struggling with is that I cannot persist my login state and access my orders.
Can someone please help me with this?

I'm using next-auth

**Comment by rodrigoddalmeida** [comment, marked as answer]
Hi @dirums, you can persist login/auth state by storing the `X-Session` header coming from the `/login` endpoint. That can be done inside your `authenticate` function in a custom `Credentials` provider.

Since Swell uses a stateful session token, it is recommended that you store the `X-Session` header value coming from all requests to Frontend API endpoints. That'll make sure changes to the user's session (like locale, current cart, etc...) are in sync.

**Reply by yodaljit** [reply]
I encountered an issue with the current code when deployed to vercel. The error says, "Failed to parse URL from null/api/account/login". Any idea why this is happening? It is working fine on localhost but not on vercel.

Here's the code for authentication:
```
import NextAuth from "next-auth"
import swell from 'swell-js'
import CredentialsProvider from "next-auth/providers/credentials"


export default NextAuth({
  providers: [
    CredentialsProvider({
        name: 'credentials',
        credentials: {
          email: { label: "Email", type: "email" },
          password: {  label: "Password", type: "password" }
        },
        async authorize(credentials, req) {
          const { email, password } = credentials
          const user = await swell.account.login(email, password)
          console.log(user)
          if (user != null){
            return user
          }

          return null
        }
    })
  ],
  callbacks: {
      async jwt({ token, user, profile, account }) {
        if(user){
        return user
        }
        return token
      },
      async session({ session, token, user }) {

        if(token){
          session.user.id = token.id
          session.user.phone = token.phone
          session.user.image = token.profile_img
          session.user.token = token.password_token
          session.user.wishlist = token.wishlist
          session.user.email_optin = token.email_optin
        }

        return session
      }
  },
  secret: "test",
  jwt: {
    encryption: true
  },
  pages: {
    signIn: '/login',
    error: '/login'
  },
  session:{
    strategy: 'jwt'
  }
})
```

**Reply by rodrigoddalmeida** [reply]
@yodaljit that error indicates that you might not have initialized the client yet. That should be done with `swell.init` and adding your credentials: https://developers.swell.is/frontend-api/setup#initialization

**Reply by yodaljit** [reply]
how is it working on localhost then?

**Reply by logeshswell** [reply]
@yodaljit `SWELL_STORE_URL` env might be missing in vercel but it would be present in your localhost. Can you confirm the same?
